pkg/agent/agent.go

```diff
@@ -67,18 +69,42 @@ type proc interface {
 	KillProcess() error
 }
 
-func New(log logging.Logger, kube kubernetes.Interface, plat platform.Platform, nodeName string) (*Agent, error) {
+func New(log logging.Logger, kube kubernetes.Interface, platform platform.Platform, nodeName string) (*Agent, error) {
 	if nodeName == "" {
 		return nil, errors.New("nodeName must be provided for Agent to manage")
 	}
 	var nodeclient corev1.NodeInterface
 	if kube != nil {
 		nodeclient = kube.CoreV1().Nodes()
+		// Determine which platform to use depending on the updater interface version
+		var node, err = nodeclient.Get(nodeName, v1meta.GetOptions{})
+		if err != nil {
+			return nil, errors.New("failed to retrieve node information")
+		}
+		// Get the updater interface version from the node label
+		var platformVersion = node.Labels[marker.UpdaterInterfaceVersionKey]
+		switch platformVersion {
+		case "2.0.0":
```

Comment:
I don't see why not, but I also don't really see why we should. The platform version here is to differentiate hosts that can support the update API or not. There's really nothing much in between.

I think ideally the label would be more explicit. Instead of a version, it would just be the supported update platform, "updog" or "update-api".